client/lib/compare-props/index.js

```diff
@@ -0,0 +1,49 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { includes, isEqual } from 'lodash';
+
+export default ( options = {} ) => {
+	const { ignore, deep, shallow } = options;
+	return ( prevProps, nextProps ) => {
```

Comment:
This sounds React-ish. Maybe name it `value`, `other` akin to Lodash?

Reply:
The purpose if this function is to be used with React props. There are many ways how to compare Javascript objects and the ambition here is not to implement some generic object comparison. For example, comparing objects with unusual prototype chains would probably not always work as expected. But for React props, it doesn't matter.